### PR TITLE
Plexon overflow fix

### LINKED ISSUE #1366: Read. plx file, error occurs when the file is larger than 2GB
**Describe the bug**
When reading. plx files, an error occurs when the file is larger than 2GB, and when the file is smaller than 1GB, it can be read and called normally.
reader= neo.io.PlexonIO(filename = file_path)
**To Reproduce**
)C:\software\Anaconda\envs\py3.9\lib\site-packages\neo\rawio\plexonrawio.py:91: RuntimeWarning: overflow encountered in long_scalars
  pos += length
C:\software\Anaconda\envs\py3.9\lib\site-packages\neo\rawio\plexonrawio.py:87: RuntimeWarning: overflow encountered in ushort_scalars
  length = bl_header['NumberOfWaveforms'] * bl_header['NumberOfWordsInWaveform'] * 2 + 16
Traceback (most recent call last):
  File "C:\Program Files\JetBrains\PyCharm 2023.3.1\plugins\python\helpers\pydev\pydevd.py", line 1527, in _exec
    pydev_imports.execfile(file, globals, locals)  # execute the script
  File "C:\Program Files\JetBrains\PyCharm 2023.3.1\plugins\python\helpers\pydev\_pydev_imps\_pydev_execfile.py", line 18, in execfile
    exec(compile(contents+"\n", file, 'exec'), glob, loc)
  File "D:\code_cloud\new_eeg_deep\scut_eeg_dl\sub_pipeline\src\sub_pipeline\pro_lfp\lfp_read.py", line 9, in <module>
    seg = neo.io.PlexonIO(filename = file_path).read_block().segments[0].events
  File "C:\software\Anaconda\envs\py3.9\lib\site-packages\neo\io\plexonio.py", line 20, in __init__
    BaseFromRaw.__init__(self, filename)
  File "C:\software\Anaconda\envs\py3.9\lib\site-packages\neo\io\basefromrawio.py", line 74, in __init__
    self.parse_header()
  File "C:\software\Anaconda\envs\py3.9\lib\site-packages\neo\rawio\baserawio.py", line 178, in parse_header
    self._parse_header()
  File "C:\software\Anaconda\envs\py3.9\lib\site-packages\neo\rawio\plexonrawio.py", line 90, in _parse_header
    block_pos[bl_type][chan_id].append(pos)
KeyError: 32769

If the error occurs when reading a file that you can't share publicly, please let us know, and we'll get in touch to discuss sharing it privately.

**Expected behaviour**
Can read. plx files normally

**Environment:**
 - Windows
 - Python version 3.9
 - Neo version    0.120 ，0.10 and nonstable version 
 - NumPy version

**Additional context**

### PR DESCRIPTION
**Pull request by h-mayorquin**
Fix #1366

### CODE REVIEW

**Review by zm711 — COMMENTED** [review]
I forgot to hit submit... Sorry for the delay.

File: neo/rawio/plexonrawio.py [hunk 1]
```diff
@@ -136,7 +136,9 @@ def _parse_header(self):
 
         while pos < data.size:
             bl_header = data[pos : pos + 16].view(DataBlockHeader)[0]
-            length = bl_header["NumberOfWaveforms"] * bl_header["NumberOfWordsInWaveform"] * 2 + 16
+            number_of_waveforms = int(bl_header["NumberOfWaveforms"])
+            number_of_words_in_waveform = int(bl_header["NumberOfWordsInWaveform"])
+            length =number_of_waveforms * number_of_words_in_waveform * 2 + 16
```
[the remark this answers is not in the document]
Comment (h-mayorquin) [reply neo/rawio/plexonrawio.py:141]: Sure : )

No idea, I guess is just the size in bytes of those blocks. It probably should done better but I don't have the bandwidth now to do this.

[the remark this answers is not in the document]
Comment (zm711) [reply neo/rawio/plexonrawio.py:141]: Cool. So I'll commit it then we can merge.

**Review by samuelgarcia — APPROVED** [review]

**Comment by samuelgarcia** [comment]
heroic!

**Comment by zm711** [comment]
Tests will fail until #1494 is merged.

**Comment by samuelgarcia** [comment]
@h-mayorquin  : I propose to merge this because the failing is because plexon2 and here you fix plexon1

**Comment by h-mayorquin** [comment]
It seems that we just needed to merge the latest changes from master. I don't see the button in the web interface, do you guys see it? I did not know that was a collaborator perk.

**Comment by zm711** [comment]
> It seems that we just needed to merge the latest changes from master. I don't see the button in the web interface, do you guys see it? I did not know that was a collaborator perk.

It's not a perk. I think it is something you have to configure. I never see it on Neo, but I don't know how to configure it.... I always just do it manually.

**Comment by h-mayorquin** [comment]
Well, so it was that, it can be merge now : )

**Pull request merged**